dicom_utils/container/input.py

```diff
@@ -65,7 +65,11 @@ def __call__(self, key: Optional[str], collection: RecordCollection, index: int,
     @classmethod
     def get_patient_id(cls, collection: RecordCollection) -> Optional[str]:
         pids = {pid for rec in collection if isinstance(rec, SupportsPatientID) and (pid := rec.PatientID) is not None}
-        return next(iter(pids)) if len(pids) == 1 else None
+        pid = next(iter(pids)) if len(pids) == 1 else None
```

Comment:
Should we log a warning if there is more than one patient ID?

Reply:
I'd rather keep that logic in a `RecordHelper` than here for a couple of reasons:
* If multiple PIDs exist this method returns `None`, which will clearly manifest to the user as output files that are not organized into a subdirectory. So it's not a hidden issue.
* If we put the warning or error handling here it is harder to override on a per-dataset basis. By using a helper we can have custom logging / assertions as needed
* In practice I don't think we'll ever encounter multiple PIDs because of how the organization pipeline groups outputs. I'm not confident enough to put an assertion here, but I can't see how it would happen during typical use.